node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/editor/CursorImpl.java

```diff
@@ -0,0 +1,365 @@
+        return this;
+    }
+
+    @Override
+    public Cursor deletePrefix() {
+        Position originalPosition = position;
+        moveToStartOfLine();
+        textBuffer.delete(position, originalPosition);
+        return this;
+    }
+
+    @Override
+    public Cursor deleteSuffix() {
+        textBuffer.delete(position, textBuffer.getEndOfLine(position));
+        return this;
+    }
+
+    @Override
+    public Cursor deleteForward() {
+        return deleteForward(1);
+    }
+
+    @Override
+    public Cursor deleteForward(int times) {
+        Position end = textBuffer.forward(position, times);
+        textBuffer.delete(position, end);
+        return this;
+    }
+
+    @Override
+    public Cursor deleteBackward() {
+        return deleteBackward(1);
+    }
+
+    @Override
+    public Cursor deleteBackward(int times) {
+        Position end = position;
+        moveBackward(times);
+        textBuffer.delete(position, end);
+        return this;
+    }
+
+    @Override
+    public Cursor deleteTo(Mark mark) {
+        Position start = mark.position();
+        Position end = position;
+        if (start.isAfter(end)) {
+            Position tmp = start;
+            start = end;
+            end = tmp;
+        }
+
+        textBuffer.delete(start, end);
+        return this;
+    }
+
+    @Override
+    public boolean replaceMatch(Pattern pattern, Function<Match, String> replacer) {
+        Optional<Match> match = moveForwardToStartOfMatch(pattern);
+        if (!match.isPresent()) {
+            return false;
+        }
+
+        // position is unaffected by delete since position == match.get().startOfMatch()
+        textBuffer.delete(match.get().startOfMatch(), match.get().endOfMatch());
+        write(replacer.apply(match.get()));
+        return true;
+    }
+
+    @Override
+    public int replaceMatches(Pattern pattern, Function<Match, String> replacer) {
+        int count = 0;
+
+        for (; replaceMatch(pattern, replacer); ++count) {
+            // empty
+        }
+
+        return count;
+    }
+
+    private void validatePosition(Position position) {
+        validateLineIndex(position.lineIndex());
+
+        int maxColumnIndex = textBuffer.getLine(position.lineIndex()).length();
+        if (position.columnIndex() < 0 || position.columnIndex() > maxColumnIndex) {
+            throw new IndexOutOfBoundsException("Column index of " + position.coordinateString() +
+                    " is not in permitted range [0," + maxColumnIndex + "]");
+        }
+    }
+
+    private void validateLineIndex(int lineIndex) {
+        int maxLineIndex = textBuffer.getMaxLineIndex();
+        if (lineIndex < 0 || lineIndex > maxLineIndex) {
+            throw new IndexOutOfBoundsException("Line index " + lineIndex +
+                    " not in permitted range [0," + maxLineIndex + "]");
+        }
+    }
+
+    private void validateMark(Mark mark) {
```

Comment:
The idea was that if the secret is verified, and the version hasn't changed, then the position MUST be valid since we made it. I could add a comment about this?

Reply:
Perhaps. I missed it :)